boot/boot.go

```diff
@@ -40,6 +42,40 @@ const (
 	TryingStatus = "trying"
 )
 
+var (
+	// EarlyRunMnt is the rootdir where ubuntu partitions are mounted during the
+	// early boot (i.e. the initramfs)
+	EarlyRunMnt = filepath.Join(dirs.GlobalRootDir, "run/mnt")
+
+	// EarlyUbuntuData is the location of ubuntu-data during early boot (i.e.
+	// the initramfs)
+	EarlyUbuntuData = filepath.Join(EarlyRunMnt, "ubuntu-data")
+
+	// EarlyUbuntuBoot is the location of ubuntu-boot during early boot (i.e.
+	// the initramfs)
+	EarlyUbuntuBoot = filepath.Join(EarlyRunMnt, "ubuntu-boot")
+
+	// EarlyUbuntuSeed is the location of ubuntu-seed during early boot (i.e.
+	// the initramfs)
+	EarlyUbuntuSeed = filepath.Join(EarlyRunMnt, "ubuntu-seed")
+
+	// EarlyWritable is the location of the writable partition during the early
+	// boot (i.e. during the initramfs)
+	EarlyWritable = filepath.Join(EarlyUbuntuData, "system-data")
+)
+
+func init() {
+	// register to change the values of Early* dir values when the global
+	// root dir changes
+	dirs.AddRootDirCallback(func(rootdir string) {
+		EarlyRunMnt = filepath.Join(rootdir, "run/mnt")
```

Comment:
this duplicates a bit the settings above, you could have a setEarlyDirs helper and then both invoke it with dirs.GlobalDirs and set it as callback